64-bit: Fix type mismatch issues.

HanJing755 opened the pull request:
bug:v/50763

Rootcause: On 64-bit platforms, some Log statements have format strings that do not match the variable types.

*Note: Please adhere to [Contributing Guidelines](https://github.com/open-vela/docs/blob/dev/CONTRIBUTING.md).*

## Summary

64-bit: Fix type mismatch issues.

## Impact

N/A.


## Testing

CI

HanJing755 committed:
64-bit: Fix type mismatch issues.

bug:v/50763

Rootcause: On 64-bit platforms, some Log statements have format strings that do not match the variable types.

Signed-off-by: v-hanjing3 <v-hanjing3@xiaomi.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/open-vela/frameworks_bluetooth?pullRequest=113) <br/>All committers have signed the CLA.

HanJing755 committed:
64-bit: Adjust the platform configuration macros in bt_config.h.

bug:v/52976

Rootcause: The configuration file of NUTTX is included after the platform macro configuration, which causes the inability to identify the compilation platform bit of Vela. To properly define the platform configuration macro, the following two adjustments are made: 1. Move the platform configuration macro to the end of bt_config.h; 2. Add compilation detection for the x86_64 platform.

Signed-off-by: v-hanjing3 <v-hanjing3@xiaomi.com>

JinxianChe commented on framework/include/bt_config.h:
```diff
@@ -130,6 +123,13 @@
 #endif
 
+// Platform: 32-bit or 64-bit
+#if defined(CONFIG_ARCH_ARM64) || defined(ARCH_X86_64) || defined(ANDROID) || (!defined(CONFIG_SIM_M32) && defined(CONFIG_ARCH_SIM))
```
add a condition (defined(CONFIG_ARCH_X86_64)).

HanJing755 commented:
start build
